protoc-gen-grpc-gateway/gengateway/template.go

```diff
@@ -210,9 +235,24 @@ var (
 	var protoReq {{.Method.RequestType.GoType .Method.Service.File.GoPkg.Path}}
 	var metadata runtime.ServerMetadata
 {{if .Body}}
-	if err := marshaler.NewDecoder(req.Body).Decode(&{{.Body.AssignableExpr "protoReq"}}); err != nil && err != io.EOF  {
+	body, berr := ioutil.ReadAll(req.Body)
```

Comment:
I'd be glad to receive recommendations on how to do this in a way that doesn't require reading the entire request body multiple times. I wrote it like this because I thought it was the most readable.